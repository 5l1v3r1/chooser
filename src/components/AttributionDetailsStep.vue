<template>
    <form class="attribution-details-form">
        <b-field label="Work Author" label-position="inside">
            <b-input
                v-model="value.creatorName"
                placeholder="John Silver"
            />
        </b-field>
        <b-field label="URL of Creator Profile" label-position="inside">
            <b-input
                v-model="value.creatorProfileUrl"
                placeholder="Creator's blog/ twitter profile/ instagram profile etc"
            />
        </b-field>
        <b-field label="Title of Work" label-position="inside">
            <b-input
                v-model="value.workTitle"
                placeholder="This work"
            />
        </b-field>
        <b-field label="Work URL" label-position="inside">
                <b-input
                    v-model="value.workUrl"
                    placeholder="https://www.example.com/image.jpg"
                />
            </b-field>
    </form>
</template>

<script>
export default {
    name: 'AttributionDetailsStep',
    props: ['value'],
    watch: {
        value() { this.$emit('input', this.$props.value) }
    }
}
</script>

<style lang="scss">
label.label {
    font-size: 0.845rem;
}
</style>
