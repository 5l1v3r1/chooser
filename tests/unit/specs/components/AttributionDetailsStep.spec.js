import { createLocalVue, mount } from '@vue/test-utils'
import Buefy from 'buefy'
import Vuex from 'vuex'
import AttributionDetailsStep from '@/components/AttributionDetailsStep'

const localVue = createLocalVue()

localVue.use(Vuex)
localVue.use(Buefy)

it('AttributionDetailsStep Component is mounted if status is current', () => {
    const wrapper = mount(AttributionDetailsStep, {
        localVue,
        propsData: {
            status: 'current'

        },
        mocks: {
            $t: key => key,
            $store: {
                state: {
                    attributionDetails: {
                        creatorName: '',
                        creatorProfileUrl: '',
                        workTitle: '',
                        workUrl: ''
                    }
                }
            }
        }

    })
    expect(wrapper.element).toMatchSnapshot()
})
