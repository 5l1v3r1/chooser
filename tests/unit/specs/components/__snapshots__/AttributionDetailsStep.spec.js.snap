// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AttributeDetailsStep Component is mounted if status is current 1`] = `
<div
  class="step-content"
>
  <div
    class="step-actions"
  >
    <p
      class="attribution-details-instructions"
    >
      
            stepper.AD.instructions
        
    </p>
     
    <form
      class="attribution-details-form"
    >
      <div
        class="field"
      >
        <label
          class="label"
        >
          stepper.AD.form.creator-name.label
        </label>
         
        <div
          class="control is-clearfix"
        >
          <input
            autocomplete="on"
            class="input"
            placeholder="stepper.AD.form.creator-name.placeholder"
            type="text"
          />
           
          <!---->
           
          <!---->
           
          <!---->
        </div>
         
        <!---->
      </div>
       
      <div
        class="field"
      >
        <label
          class="label"
        >
          stepper.AD.form.creator-profile.label
        </label>
         
        <div
          class="control is-clearfix"
        >
          <input
            autocomplete="on"
            class="input"
            placeholder="stepper.AD.form.creator-profile.placeholder"
            type="text"
          />
           
          <!---->
           
          <!---->
           
          <!---->
        </div>
         
        <!---->
      </div>
       
      <div
        class="field"
      >
        <label
          class="label"
        >
          stepper.AD.form.work-title.label
        </label>
         
        <div
          class="control is-clearfix"
        >
          <input
            autocomplete="on"
            class="input"
            placeholder="stepper.AD.form.work-title.placeholder"
            type="text"
          />
           
          <!---->
           
          <!---->
           
          <!---->
        </div>
         
        <!---->
      </div>
       
      <div
        class="field"
      >
        <label
          class="label"
        >
          stepper.AD.form.work-url.label
        </label>
         
        <div
          class="control is-clearfix"
        >
          <input
            autocomplete="on"
            class="input"
            placeholder="stepper.AD.form.work-url.placeholder"
            type="text"
          />
           
          <!---->
           
          <!---->
           
          <!---->
        </div>
         
        <!---->
      </div>
    </form>
  </div>
</div>
`;

exports[`AttributionDetailsStep Component is mounted if status is current 1`] = `
<div
  class="step-content"
>
  <div
    class="step-actions"
  >
    <p
      class="attribution-details-instructions"
    >
      
            stepper.AD.instructions
        
    </p>
     
    <form
      class="attribution-details-form"
    >
      <div
        class="field"
      >
        <label
          class="label"
        >
          stepper.AD.form.creator-name.label
        </label>
         
        <div
          class="control is-clearfix"
        >
          <input
            autocomplete="on"
            class="input"
            placeholder="stepper.AD.form.creator-name.placeholder"
            type="text"
          />
           
          <!---->
           
          <!---->
           
          <!---->
        </div>
         
        <!---->
      </div>
       
      <div
        class="field"
      >
        <label
          class="label"
        >
          stepper.AD.form.creator-profile.label
        </label>
         
        <div
          class="control is-clearfix"
        >
          <input
            autocomplete="on"
            class="input"
            placeholder="stepper.AD.form.creator-profile.placeholder"
            type="text"
          />
           
          <!---->
           
          <!---->
           
          <!---->
        </div>
         
        <!---->
      </div>
       
      <div
        class="field"
      >
        <label
          class="label"
        >
          stepper.AD.form.work-title.label
        </label>
         
        <div
          class="control is-clearfix"
        >
          <input
            autocomplete="on"
            class="input"
            placeholder="stepper.AD.form.work-title.placeholder"
            type="text"
          />
           
          <!---->
           
          <!---->
           
          <!---->
        </div>
         
        <!---->
      </div>
       
      <div
        class="field"
      >
        <label
          class="label"
        >
          stepper.AD.form.work-url.label
        </label>
         
        <div
          class="control is-clearfix"
        >
          <input
            autocomplete="on"
            class="input"
            placeholder="stepper.AD.form.work-url.placeholder"
            type="text"
          />
           
          <!---->
           
          <!---->
           
          <!---->
        </div>
         
        <!---->
      </div>
    </form>
  </div>
</div>
`;
